lib/realtime/tenants/scheduled_message_cleanup.ex

```diff
@@ -0,0 +1,95 @@
+defmodule Realtime.Tenants.ScheduledMessageCleanup do
+  @moduledoc """
+  Scheduled tasks for the Tenants.
+  """
+
+  use GenServer
+  require Logger
+
+  import Ecto.Query
+  import Realtime.Helpers, only: [log_error: 2]
+
+  alias Realtime.Api.Tenant
+  alias Realtime.Database
+  alias Realtime.Messages
+  alias Realtime.Nodes
+  alias Realtime.Repo
+
+  defstruct timer: nil, region: nil, chunks: nil
+
+  def start_link(_args) do
+    timer = Application.get_env(:realtime, :schedule_clean, :timer.hours(5))
+    region = Application.get_env(:realtime, :region)
+    chunks = Application.get_env(:realtime, :chunks, 10)
+    state = %__MODULE__{timer: timer, region: region, chunks: chunks}
+    GenServer.start_link(__MODULE__, state, name: __MODULE__)
+  end
+
+  @impl true
+  def init(%__MODULE__{timer: timer} = state) do
+    Process.send_after(self(), :delete_old_messages, timer)
```

Comment:
Maybe need to try removing it when the process starts and then after a timeout. For example, tables will not be cleared if the node's lifetime is less than the timeout (in case of rebooting or an incident, etc)

Reply:
this is in a way intended, when we have the node starting this might increase the load on boot which can impact it (it's during boot that we see most load / created processes as expected)

we can find other ways, e.g. start after 10 minutes (aka have a new config for "scheduled_start_after")

Reply:
agreed, "scheduled_start_after" is nice